engine/univNames.ml

```diff
@@ -65,8 +60,11 @@ let universe_binders_with_opt_names orig names =
       CErrors.user_err ~hdr:"universe_binders_with_opt_names"
         Pp.(str "Universe instance should have length " ++ int len)
   in
-  let fold i acc na = match na with
-  | Name id -> Names.Id.Map.add id (Level.var i) acc
-  | Anonymous -> acc
+  let fold i acc na =
+    let id = match na with
+      | Name id -> id
+      | Anonymous -> Id.of_string_soft ("#"^string_of_int i)
```

Comment:
No sure this is the best name we can produce, but getting something non conflicting with another name would need more code change.